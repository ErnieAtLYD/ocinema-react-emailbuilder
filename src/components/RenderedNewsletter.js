// @flow
import React from "react";
import LayoutItem from "../components/LayoutItem";

type RenderedNewsletterType = {
  layout: NewsletterLayoutType,
  deleteLayoutItem: Function,
  dropElementIntoColumnContent: Function,
  dropElementIntoColumnElement: Function,
  duplicateLayoutItem: Function,
  editLayoutItem: Function,
  moveLayoutItem: Function,
};

const RenderedNewsletter = ({
  layout,
  deleteLayoutItem,
  dropElementIntoColumnContent,
  dropElementIntoColumnElement,
  duplicateLayoutItem,
  editLayoutItem,
  moveLayoutItem,
}: RenderedNewsletterType): any =>
  layout.map((item: NewsletterLayoutItemTypeV2, index: number): React$Element<
    typeof LayoutItem
  > => (
    <LayoutItem
      key={item.id}
      id={item.id}
      index={index}
      item={item}
      deleteLayoutItem={deleteLayoutItem}
      dropElementIntoColumnContent={dropElementIntoColumnContent}
      dropElementIntoColumnElement={dropElementIntoColumnElement}
      duplicateLayoutItem={duplicateLayoutItem}
      editLayoutItem={editLayoutItem}
      moveLayoutItem={moveLayoutItem}
    />
  ));

export default RenderedNewsletter;
