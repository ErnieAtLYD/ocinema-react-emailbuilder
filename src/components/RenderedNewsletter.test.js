import React from 'react';
import { shallow } from 'enzyme';
import RenderedNewsletter from './RenderedNewsletter'

describe('RenderedNewsletter', () => {

  it('renders layout items', () => {
    const layout = [{id: 1}, {id: 2}];
    const wrapper = shallow(<RenderedNewsletter layout={layout} />);
    // console.log(wrapper.debug());
    expect(wrapper.find('LayoutItem')).toBeDefined();
    expect(wrapper.find('LayoutItem')).toHaveLength(layout.length);
    expect(wrapper).toMatchSnapshot();
  });
});
