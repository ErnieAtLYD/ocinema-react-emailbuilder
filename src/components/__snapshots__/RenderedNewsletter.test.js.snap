// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderedNewsletter renders layout items 1`] = `
Array [
  <LayoutItem
    id={1}
    index={0}
    item={
      Object {
        "id": 1,
      }
    }
    key="1"
  />,
  <LayoutItem
    id={2}
    index={1}
    item={
      Object {
        "id": 2,
      }
    }
    key="2"
  />,
]
`;
